### DISCUSSION TITLE
how to use a joinhandle twice in a select?

### DISCUSSION THREAD

**Discussion by dan-da** [tokio-rs/tokio#6969] [Q&A]
Let's say I spawn a task A and I have a channel B.   I want to loop and select!() either that A finishes or a quit message is received on B.   If the latter, then abort() task A.

something iike: 

```
    tokio::select!{
        Ok(_v) = a_joinhandle => { ... },
        
        _ = b_rx.recv() => {
            a_joinhandle.abort();
        }
    }
```

Unfortunately I get a compiler error at the abort call because a_joinhandle is already moved above.

So, how to accomplish this?

here's a [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5f063a08f7e21ef005af5ec1aa68bffa)

**Comment by dan-da** [comment]
answering myself.   I found the solution [here](https://github.com/tokio-rs/tokio/discussions/4019#discussioncomment-1123333).

```
Ok(_v) = &mut a_joinhandle => { ... },
```

and voila it works.

**Reply by Darksonn** [reply]
Yep. That's the way to do it.
